homeassistant/components/aladdin_connect/__init__.py

```diff
@@ -1,48 +1,53 @@
-"""The aladdin_connect component."""
+"""The Aladdin Connect Genie integration."""
 
-import logging
-from typing import Final
-
-from AIOAladdinConnect import AladdinConnectClient
-import AIOAladdinConnect.session_manager as Aladdin
-from aiohttp import ClientError
+from __future__ import annotations
 
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.const import Platform
 from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
-
-from .const import CLIENT_ID, DOMAIN
+from homeassistant.helpers import aiohttp_client, config_entry_oauth2_flow
 
-_LOGGER: Final = logging.getLogger(__name__)
+from . import api
+from .const import CONFIG_FLOW_MINOR_VERSION, CONFIG_FLOW_VERSION
 
-PLATFORMS: list[Platform] = [Platform.COVER, Platform.SENSOR]
+PLATFORMS: list[Platform] = [Platform.COVER]
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up platform from a ConfigEntry."""
-    username = entry.data[CONF_USERNAME]
-    password = entry.data[CONF_PASSWORD]
-    acc = AladdinConnectClient(
-        username, password, async_get_clientsession(hass), CLIENT_ID
+    """Set up Aladdin Connect Genie from a config entry."""
+    implementation = (
+        await config_entry_oauth2_flow.async_get_config_entry_implementation(
+            hass, entry
+        )
```

Comment:
Should we be doing all the intermediate assignments for single use?
Remove `implementation` and `session` and inline the code?

Reply:
Because they are long I don't mind this tbh, makes it a bit cleaner

Reply:
I too like the easier read, but I think it's a startup performance hit that bdraco would go after :) (and has on several of my integrations)

Reply:
Assigning to a variable in a temporary context isn't that bad tbf, I use it in Withings as well. Curious to those PRs of bdraco though!

Reply:
it's based on the scaffold template.